README.md

```diff
@@ -169,23 +188,26 @@ do {
 5. Handle the `doChallenge` flow: 
 
 - If the Authentication response that is returned indicates that the Challenge Flow must be applied, the 3DS Requestor App calls the `doChallenge` method with the required input `ChallengeParameters`. The `doChallenge` method initiates the challenge process.
-- The `doChallenge` function now returns a `ChallengeResult` object with `transactionStatus` and `sdkTransactionID`.
+- The `doChallenge` function now returns `Result` type with two cases
+  - `AuthenticationResult` with `transactionStatus` and `sdkTransactionID`  👉&nbsp;&nbsp;[More info about transaction status](https://www.checkout.com/docs/business-operations/use-the-dashboard/payment-activity/track-3ds-events#Transaction_status)
+  - `AuthenticationError` with a `message`.
+
+
 ```swift
 let params = ChallengeParameters(threeDSServerTransactionID: response.transactionId,
                                  acsTransactionID: response.acs.transactionId,
                                  acsRefNumber: response.acs.referenceNumber,
                                  acsSignedContent: response.acs.signedContent)
 transaction?.doChallenge(challengeParameters: params, completion: { [weak self]  result in
-     guard let self else { return }
      switch result {
-      case .success(let result):
-          // handle success scenario
+      case .success(let authenticationResult):
+         // handle authentication result. Checkout Payment Authorisation section.
       case .failure(let error):
          // handle failure scenario
        }
       // call close and cleanUp methods after challenge flow is completed. 
-      self.transaction?.close()
-      self.standalone3DSService?.cleanUp()
+      self?.transaction?.close()
+      self?.standalone3DSService?.cleanUp()
```

Comment:
nit: I think it's better practice to keep a strong reference to self after it being weakified into the closure. We had some discussions around this with Alex before he left but can discuss again in our next meeting. If it's a strong reference, then we get to make sure that these two lines either run together or fail together. Currently, in a theoretical case of self getting deallocated in between these two lines, program gets half run and might go into an unknown state. Not too important though.